### ISSUE TITLE
Cannot add Remmina to Gnome favorites

### ISSUE THREAD

**Issue by angus73** [FreeRDP/Remmina#1447]
I am not able to add Remmina to the Gnome favorites list. If I run it and right-click on the icon in the dock, the menu (which should contain the "add to favorites" item) does not show up. Instead, only the name of the application is displayed.

Other applications (i.e. firefox, libreoffice) allow me to do it.

Running under Gnome Shell version 3.24.3
Remmina - 1.2.0-rcgit-26 (git rcgit-26)
Fedora 26

see also: https://ask.fedoraproject.org/en/question/116350/cannot-add-remmina-to-favourites/

**Comment by antenore** [comment]
Can you check if fedora installs the remmina.desktop file?

It must be named exactly as the remmina binary name + .desktop, for example remmina.desktp

We probably name it org.remmina.Remmina.desktop for other reasons I don't remember, if it's the case do a copy of the file as I've mentioned to you.

BTW, for me this is more a packaging and configuration issue than a Remmina bug, it's hard to deliver desktop files and icons that works on every distro and ecosystem of the word :-(

Let me know

**Comment by angus73** [comment]
Assuming I am looking in the right place:

$ ls -l /usr/share/applications/ |grep -i remmina
-rw-r--r--. 1 root root  5096  5 gen 00.24 org.remmina.Remmina.desktop
-rw-r--r--. 1 root root   232  5 gen 00.24 remmina-file.desktop

should I provide their contents?

Thanks
Andrea

**Comment by antenore** [comment]
No, just copy org.remmina.Remmina.desktop to remmina.desktop and restart gnome.

**Comment by angus73** [comment]
ok it works! Thank you very much.
I think it's safe to delete the old ones isn't it?

**Comment by antenore** [comment]
Not at all, we have put in place the other one to be compliant with the freedesktop standards, so if you remove it you will have troubles somewhere else and it could be that you already have troubles because you have both and in any case during the next upgrade the file will be created again.

I'll investigate further on these files when I'll have time.

Keep both and come back to this thread if you'll suffer from other (related) issues, I'm really interested.

**Comment by angus73** [comment]
Thank you for your answer.

Well, actually I have now three of them:

$ ls -l /usr/share/applications/ |grep -i remmina
-rw-r--r--. 1 root root  5096  5 gen 00.24 org.remmina.Remmina.desktop
-rw-r--r--. 1 root root  5096 18 gen 09.53 remmina.desktop
-rw-r--r--. 1 root root   232  5 gen 00.24 remmina-file.desktop

and this leads up to two "Remmina" menu items in the Gnome Applications menu (Internet section)

Maybe I can get rid of the remmina-file.desktop?

**Comment by antenore** [comment]
That one is used to connect to a remmina profile double clickcking a remmina file ($HOME/.local/share/remmina/").

It should be the org* file that cause your issue.

I don't know what to answer you honestly, it's somehow a freedesktop/gnome/setup issue and could be also a Remmina problem but I lack (yet) the knowledge.

I've followed what is written here:

https://standards.freedesktop.org/desktop-entry-spec/latest/ar01s02.html
https://developer.gnome.org/integration-guide/stable/desktop-files.html.en

**Comment by PaulMHalliwell** [comment]
I've renamed my org.remmina.desktop to a .bak extension in order to remove the second icon in GNOME, and so far it seems to be stable. 

I presume that it may cause issues, or simply get re-created when updating via dnf, if so you can always revert it back to being a .desktop file and run the update again.

**Comment by der-eismann** [comment]
I had the same problem on Solus, all we had to do to fix it was add `StartupWMClass=remmina` to the file `org.remmina.Remmina.desktop`. Afterwards it was "pin-able" to the taskbar.

**Comment by giox069** [comment]
Thank you @der-eismann: `StartupWMClass=remmina` seems to be the right solution. I'm committing it now in the next branch, it will be available with the next release (1.2.0-rcgit.28). So for all people interested in having a better Gnome shell launcher support before -rcgit.28, I can suggest to manually edit `/usr/share/applications/org.remmina.Remmina.desktop` and `StartupWMClass=remmina` manually inside the `[Desktop entry]' section.
